FindPeakElement.java

```diff
@@ -20,18 +21,21 @@
      * @return peak element index
      */
     public int findPeakElement(int[] nums) {
-        int low = 0, high = nums.length - 1;
+        if(nums == null || nums.length == 0) {
+            return -1;
+        }
+        if(nums.length == 1) {
+            return 0;
+        }
+        int low = 0, high = nums.length- 1;
         while (low < high) {
             int mid = low + (high - low) / 2;
-            if ((mid > 0 && nums[mid] > nums[mid - 1]) && (mid < nums.length - 1 && nums[mid] > nums[mid + 1])) {
```

Comment:
how can you return mid whithout checking whether the value at mid = mid+2 or not?

Reply:
This is based on the template 2 of binary search that Komal explained in the previous session
Where we do not have a condition check in the while loop and return when the loop terminates.

I didnt understand this part `mid = mid+2` can you please explain?

Reply:
I think value at mid = mid + 2 is the condition that in the problem of finding the missing element in the sorted array here: 
https://github.com/super30admin/Competitive-Coding-1/pull/19